### PR TITLE
Ghidra function import script

### PR DESCRIPTION
**Pull request by AlexApps99**
The Ghidra import script will do automatic demangling, and will set tags based on the decompilation status of the function.
If run multiple times it will update the tags and function names as they are changed.

I also plan to release a script that will import DWARF information and potentially RTTI, once I am satisfied with it. (in another PR)

~~I also added a fix for a build warning~~, and added Ghidra's `*.lock` files to the `.gitignore`.

---
This change is [Reviewable](https://reviewable.io/reviews/zeldaret/botw/52)

### CODE REVIEW

**Comment by AlexApps99** [comment]
It looks like the build bot failed due to some kind of issue with the Rust tooling

File: src/KingSystem/Resource/Actor/resResourceAnimInfo.cpp [hunk 1]
```diff
@@ -106,7 +106,7 @@ bool AnimInfo::parse_(u8* data, size_t size, sead::Heap* heap) {
 #else
             info->entries = static_cast<SwordBlur*>(
                 heap->tryAlloc(sizeof(SwordBlur) * info->num_entries, alignof(SwordBlur)));
-            std::memset(info->entries, 0, sizeof(SwordBlur) * info->num_entries);
+            std::memset((void*)info->entries, 0, sizeof(SwordBlur) * info->num_entries);
```
Comment (leoetlino) [inline src/KingSystem/Resource/Actor/resResourceAnimInfo.cpp:109]: This should be in a separate PR

Comment (AlexApps99) [reply src/KingSystem/Resource/Actor/resResourceAnimInfo.cpp]: Done

**Review by leoetlino — APPROVED** [review]
